cmd/kubeadm/app/util/version.go

```diff
@@ -132,8 +137,8 @@ func splitVersion(version string) (string, string, error) {
 }
 
 // Internal helper: return content of URL
-func fetchFromURL(url string) (string, error) {
-	client := &http.Client{Transport: netutil.SetOldTransportDefaults(&http.Transport{})}
+func fetchFromURL(url string, timeout time.Duration) (string, error) {
```

Comment:
I'd love to see a test here (passing in the duration makes that a bit easier too)